### PR TITLE
fix(content): Earth day jobs no longer offer unless you can land on Earth

### PR DESCRIPTION
**Pull request by ziproot**
**Bug fix**

This PR addresses the bug described in issue #7663 

## Summary
This adds a to offer condition that only makes the job offer if you can land on Earth. This requires #9716 so it is marked as draft until that PR is merged.

## Testing Done
Tested to make sure no Earth Day jobs offer.

## Save File
This save file can be used to test these changes: 
[Test Pilot~Earth Day.txt](https://github.com/endless-sky/endless-sky/files/14027090/Test.Pilot.Earth.Day.txt)
Depart and land several times with this PR and with the stable branch. See how this PR fixes the issue with #9716 and stable doesn't.

### CODE REVIEW

**Comment by ziproot** [comment]
This should work. I have finally outsmarted the game.

**Review by alexrovw — CHANGES_REQUESTED** [review]
I agree with changing this behavior, but I think there's a better way to implement it.

**Concerns with the proposed solution**:
1. Hacky: Tricking the filter is a bit hacky, and it isn't self-documenting. Someone in the future would need to know why you instituted this workaround, or they might think it's bad code and change it.
2. Fragile: If there is ever a change that either removes/replaces `urban` from Earth or adds a new landable to Sol that has `urban`, this change would result in a bug.

**Alternative proposal**:

I suggest that this would be better resolved with the `to offer` block.
Something like
```
to offer
	"reputation: Republic" > 0
```
was brought up in #7663, but @ThrawnCA correctly pointed out that this would prevent players whom have tributed Earth from seeing the jobs.
So why don't we just add that as a condition?

```
to offer
	or
		"reputation: Republic" > 0
		has "tribute: Earth"
```
The above is what I propose. It would be more resilient and self-documenting than the filter hack. Would you consider implementing that instead?

**Comment by ziproot** [comment]
I have given earth a special attribute instead. This way in all cases, if you can land on Earth, you will be offered the mission. If you cannot land on Earth, you will not be offered the mission.

If Earth has required reputation -10, then you will not be offered Earth Day jobs to Earth if your reputation is between -10 and 0 even though all other jobs would offer to Earth.

If you can get required reputation as an autocondition I will switch to a `to offer` condition. If the location filter doesn't turn planet X into a fixed destination (my preferred option) I will switch to that. For now, this is the best way to ensure consistency with other missions and jobs. Yes, it's a bit confusing, but it's the fact that fixed destinations and some location filters work one way, and every other location filter works a different way, that's confusing.

File: data/map planets.txt [hunk 1]
```diff
@@ -1095,7 +1095,8 @@ planet "Dwelling of Speloog"
 	security 0.4
 
 planet Earth
-	attributes factory "near earth" research tourism urban
+	# Earth attribute is required for earth day job location filters to filter out planets that you cannot land on. A better option is for "planet Earth" to not be replaced with a fixed destination.
+	attributes factory "near earth" research tourism urban earth
```
Comment (alexrovw) [inline data/map planets.txt:1099]:
```suggestion
	attributes earth factory "near earth" research tourism urban
```
Attributes need to be in alphabetical order

File: data/map planets.txt [hunk 2]
```diff
@@ -1095,7 +1095,8 @@ planet "Dwelling of Speloog"
 	security 0.4
 
 planet Earth
-	attributes factory "near earth" research tourism urban
+	# Earth attribute is required for earth day job location filters to filter out planets that you cannot land on. A better option is for "planet Earth" to not be replaced with a fixed destination.
```
Comment (alexrovw) [inline data/map planets.txt:1098]:
```suggestion
	# "earth" attribute is used for "Earth Day" jobs as a workaround to fail the location filter when Earth is hostile.
```
Less wordy

File: data/human/near earth jobs.txt [hunk 3]
```diff
@@ -127,7 +133,10 @@ mission "To Earth Day celebration [2]"
 	source
 		near "Sol" 1 15
 		government "Republic" "Free Worlds" "Syndicate" "Neutral"
-	destination "Earth"
+	destination
+		system "Sol"
+		# Replace with planet Earth and remove attribute from Earth when doing so would not result in the game turning it into a fixed destination. This way Earth is filtered out if you cannot land there.
```
Comment (alexrovw) [inline data/human/near earth jobs.txt:138]:
```suggestion
		# "earth" attribute is used for "Earth Day" jobs as a workaround to fail the location filter when Earth is hostile.
```

File: data/human/near earth jobs.txt [hunk 4]
```diff
@@ -92,7 +95,10 @@ mission "To Earth Day celebration [1]"
...
```
Comment (alexrovw) [inline data/human/near earth jobs.txt:100]:
```suggestion
		# "earth" attribute is used for "Earth Day" jobs as a workaround to fail the location filter when Earth is hostile.
```

File: data/human/near earth jobs.txt [hunk 5]
```diff
@@ -58,7 +58,10 @@ mission "To Earth Day celebration [0]"
...
```
Comment (alexrovw) [inline data/human/near earth jobs.txt:63]:
```suggestion
		# "earth" attribute is used for "Earth Day" jobs as a workaround to fail the location filter when Earth is hostile.
```

**Comment by Amazinite** [comment]
This seems more hacky to me.

**Review by alexrovw — CHANGES_REQUESTED** [review]
I've proposed some changes to the comments to improve readability, but I'm not 100% sold on using this as a workaround. It's still reliant on the location filter behavior, but at least there are comments to explain the situation.

Personally, I think it would be valid for Earth Day jobs to disappear from the job board _in any scenario when your Republic reputation is < 0; nobody's going to want to fly to Earth in a ship that's going to be shot at by the Republic. I'm somewhat comfortable with allowing these jobs to reappear if you dominate Earth, though.

I think your point about the required reputation is true, but the situation in which Earth's required reputation would be modified to be negative is pretty unlikely. My ideal fix would be to have some kind of `"can land on: <planet>"` attribute for situations like this, but that's outside the scope of this PR.

**Comment by ziproot** [comment]
Alright, I have changed it. I wouldn't have to resort to hacks or half measures (if Earth isn't controlled by the Republic this breaks) if the location filter of planet "Earth" didn't turn into a fixed destination.

**Comment by Amazinite** [comment]
Give #9716 a spin.

**Comment by ziproot** [comment]
I gave it a spin. It works!

**Comment by ziproot** [comment]
~~dev abuse~~
image

**Pull request merged**